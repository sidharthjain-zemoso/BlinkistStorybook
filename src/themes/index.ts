import { createTheme } from "@mui/material/styles";

declare module "@mui/material/Typography" {
  interface TypographyPropsVariantOverrides {
    heading: true;
    heading3: true;
    subtitle3: true;
    body3: true;
    caption1: true;
    caption2: true;
    caption3: true;
  }
}

declare module "@mui/material/styles" {
  interface TypographyVariants {
    heading: React.CSSProperties;
    heading3: React.CSSProperties;
    subtitle3: React.CSSProperties;
    body3: React.CSSProperties;
    caption1: React.CSSProperties;
    caption2: React.CSSProperties;
    caption3: React.CSSProperties;
  }

  interface TypographyVariantsOptions {
    heading?: React.CSSProperties;
    heading3: React.CSSProperties;
    subtitle3?: React.CSSProperties;
    body3?: React.CSSProperties;
    caption1?: React.CSSProperties;
    caption2?: React.CSSProperties;
    caption3: React.CSSProperties;
  }

  interface PaletteColor {
    darker?: string;
    border?: string;
    placeholder?: string;
  }

  interface SimplePaletteColorOptions {
    darker?: string;
    placeholder?: string;
    border?: string;
  }

  interface Palette {
    greyscale: Palette["primary"];
    textcolor: Palette["primary"];
    backgroundcolor: Palette["primary"];
  }

  interface PaletteOptions {
    greyscale?: PaletteOptions["primary"];
    textcolor?: PaletteOptions["primary"];
    backgroundcolor?: PaletteOptions["primary"];
  }
}

export const theme = createTheme({
  components: {
    MuiButtonBase: {
      styleOverrides: {
        root: {
          // "&:hover": {
          //   backgroundColor: "none",
          // },
        },
      },
    },
  },
  spacing: 4,
  palette: {
    primary: {
      light: "#E2F2E9",
      main: "#2CE080",
      dark: "#22C870",
      darker: "#20BA68",
    },
    secondary: {
      light: "#DFE8F6",
      main: "#0365F2",
    },
    greyscale: {
      light: "#BAC9CF",
      main: "#6C787F",
      dark: "#3A4649",
      darker: "#042330",
    },
    textcolor: {
      light: "#6D787E",
      main: "#03314B",
      placeholder: "#747575",
      dark: "#042330",
    },
    backgroundcolor: {
      light: "#F7F7F7",
      main: "#F1F6F4",
    },
  },
  typography: {
    heading: {
      fontSize: "36px",
      lineHeight: "45px",
      // fontWeight: 700,
      fontFamily: "Cera Pro Medium",
    },
    heading3: {
      fontSize: "24px",
      lineHeight: "30.17px",
      // fontWeight: 700,
      fontFamily: "Cera Pro Medium",
    },
    subtitle1: {
      fontSize: "24px",
      lineHeight: "32px",
      // fontWeight: 500,
      fontFamily: "Cera Pro Medium",
    },
    subtitle2: {
      fontSize: "18px",
      lineHeight: "24px",
      fontWeight: 700,
      fontFamily: "Cera Pro Light",
    },
    subtitle3: {
      fontSize: "18px",
      lineHeight: "24px",
      // fontWeight: 400,
      fontFamily: "Cera Pro Light",
    },
    body1: {
      fontSize: "16px",
      lineHeight: "20px",
      // fontWeight: 700,
      fontFamily: "Cera Pro Medium",
    },
    body2: {
      fontSize: "16px",
      lineHeight: "24px",
      // fontWeight: 400,
      fontFamily: "Cera Pro Light",
    },
    body3: {
      fontSize: "22px",
      lineHeight: "20px",
      // fontWeight: 400,
      fontFamily: "Cera Pro Light",
    },
    caption1: {
      fontSize: "14px",
      lineHeight: "22px",
      // fontWeight: 400,
      fontFamily: "Cera Pro Light",
    },
    caption2: {
      fontSize: "14px",
      lineHeight: "22px",
      // fontWeight: 700,
      fontFamily: "Cera Pro Medium",
    },
    caption3: {
      fontSize: "14px",
      lineHeight: "17.6px",
      // fontWeight: 400,
      fontFamily: "Cera Pro Light",
    },
  },
});

// theme.overrides = {
//   ...theme.overrides,
//   MuiButton: {
//     root: {
//     }
//   }
// }
