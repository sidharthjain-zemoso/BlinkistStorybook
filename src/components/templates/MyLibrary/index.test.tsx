import { render, screen, fireEvent, getByTestId } from "@testing-library/react";
import MyLibraryTemplate from "./";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "@mui/styles";
import { theme } from "../../../themes";

const DummyView = () => <div>Dummy View</div>;

const MockTemplate = () => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <MyLibraryTemplate
          header={<DummyView />}
          heading={<DummyView />}
          tabs={<DummyView />}
          footer={<DummyView />}
        />
      </ThemeProvider>
    </BrowserRouter>
  );
};

describe("MyLibrary Template", () => {
  it("Render MyLibrary Template", () => {
    render(<MockTemplate />);
    const template = screen.getByTestId("my-library-template-id");
    expect(template).toBeDefined();
    expect(template).toMatchSnapshot();
  });
});
