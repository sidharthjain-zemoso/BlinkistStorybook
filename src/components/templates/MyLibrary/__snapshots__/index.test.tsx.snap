// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyLibrary Template Render MyLibrary Template 1`] = `
<div
  class="my-library-template-container"
  data-testid="my-library-template-id"
>
  <div
    class="container"
  >
    <div>
      Dummy View
    </div>
    <div
      class="makeStyles-contentContainer-1 MuiBox-root css-0"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-direction-xs-column css-vacfyj-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div>
            Dummy View
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div>
            Dummy View
          </div>
        </div>
      </div>
    </div>
    <div>
      Dummy View
    </div>
  </div>
</div>
`;
